tools/c7n_index/c7n_index/metrics.py

```diff
@@ -544,10 +575,53 @@ def index_resources(
                 account['name'], region, policy['name']))
 
 
+def index_sqs_msgs(config, queue_msg):
+    msg_json = json.loads(zlib.decompress(base64.b64decode(queue_msg['Body'])))
+    indexer = get_indexer(config)
+
+    # Reformat tags for ease of index/search
+    for resource in msg_json['resources']:
+        if 'Tags' in resource and len(resource['Tags']) != 0:
+            resource['Tags'] = {
+                t['Key']: t['Value'] for t in resource['Tags']}
+    indexer.index_sqs(queue_msg=msg_json)
+
+
+@cli.command(name='sqs-consumer')
+@click.option('-c', '--config', required=True, help="Config file")
+@click.option('-u', '--url', required=True, help="Queue URL")
+@click.option('--concurrency', default=5)
+@click.option('--verbose/--no-verbose', default=False)
+def sqs_consumer(config, url,concurrency, verbose=False):
+    """Receive messages from SQS -> push to ES"""
+    logging.basicConfig(level=(verbose and logging.DEBUG or logging.INFO))
+    logging.getLogger('botocore').setLevel(logging.WARNING)
+    logging.getLogger('elasticsearch').setLevel(logging.WARNING)
+
+    # validating config.
+    with open(config) as fh:
+        config = yaml.safe_load(fh.read())
+    jsonschema.validate(config, CONFIG_SCHEMA)
+
+    # setup variables
+    region = url.split('.')[1]
+    client = boto3.Session(region_name=region).client('sqs')
+    msg_iterator = sqsexec.MessageIterator(client=client, queue_url=url)
+
+    try:
+        with ProcessPoolExecutor(max_workers=concurrency) as w:
+            futures = []
+
+            for msg in msg_iterator:
+                futures.append(w.submit(index_sqs_msgs(config=config, queue_msg=msg)))
```

Comment:
This syntax is incorrect for submit.

you need to change it to `w.submit(index_sqs_msgs, config, msg)`

Reply:
thanks. Fixed.